cfme/configure/settings.py

```diff
@@ -20,154 +22,189 @@ class TimeProfileAddForm(View):
+        return new_time_profile
+
+
+@attr.s
+class TimeProfileCollection(BaseCollection):
+
+    ENTITY = TimeProfile
+
+    def create(self, description=None, scope=None, days=True, hours=True,
+               timezone=None, cancel=False):
+        """
+        Args:
+            description (str): It's the descriptive name of the time_profile.
+            scope: It's the option 'All User' or 'Current User' from dropdown.
+            days (bool): It's the option to switch on or switch off the days Bootstrap switch.
+            hours (bool): It's the option to switch on or switch off the hours Bootstrap switch.
+            timezone: It's the required Time Zone for the time_profile.
+            cancel (bool) : It's a flag used to cancel or not the create operation.
+
+        return: It returns the object of the newly created time_profile object.
+        """
+
+        time_profile = self.instantiate(description, scope, days, hours, timezone)
+
+        view = navigate_to(self, 'Add')
+        view.form.fill({
+            'description': description,
+            'scope': scope,
+            'days': days,
+            'hours': hours,
+            'timezone': timezone,
+        })
+        if not cancel:
+            view.form.add.click()
+            view.flash.assert_no_error()
+        return time_profile
+
+    def delete(self, cancel=False, *time_objs):
+        """
+        This method performs the delete operation.
+
+        Args:
+            cancel (bool) : It's a flag used for selecting Ok or Cancel from delete confirmation
+             dialogue box
+            time_objs : It's time profile object.
+        """
 
-    def delete(self):
         view = navigate_to(self, 'All')
-        rows = view.table
-        for row in rows:
-            if row.description.text == self.description:
-                row[0].check()
-        view.configuration.item_select("Delete selected Time Profiles", handle_alert=True)
+        for time_obj in time_objs:
+            view.entities.table.row(Description=time_obj.description)[0].check()
+        view.configuration.item_select("Delete selected Time Profiles",
+                                       handle_alert=not cancel)
+        view.flash.assert_no_error()
 
 
-@navigator.register(Timeprofile, 'All')
-class TimeprofileAll(CFMENavigateStep):
+@navigator.register(TimeProfileCollection, 'All')
+class TimeProfileCollectionAll(CFMENavigateStep):
     VIEW = TimeProfileAllView
     prerequisite = NavigateToAttribute('appliance.server', 'MySettings')
 
     def step(self):
-        self.prerequisite_view.tabs.time_profile.select()
+        self.view.mysetting.tabs.time_profile.select()
 
 
-@navigator.register(Timeprofile, 'Add')
-class TimeprofileAdd(CFMENavigateStep):
-    VIEW = TimeProfileAddFormView
+@navigator.register(TimeProfileCollection, 'Add')
+class TimeProfileAdd(CFMENavigateStep):
+    VIEW = TimeProfileView
     prerequisite = NavigateToSibling('All')
 
-    def am_i_here(self):
```

Comment:
**REQUIRED**: Please don't remove these, they are important. Add/Edit forms, should always be navigated to fresh